mythril/laser/ethereum/instructions.py

```diff
@@ -1184,18 +1186,30 @@ def extcodecopy_(self, global_state: GlobalState) -> List[GlobalState]:
             global_state=global_state,
         )
 
-    @StateTransition
+    @StateTransition()
     def extcodehash_(self, global_state: GlobalState) -> List[GlobalState]:
         """
 
         :param global_state:
         :return: List of global states possible, list of size 1 in this case
         """
-        # TODO: To be implemented
-        address = global_state.mstate.stack.pop()
-        global_state.mstate.stack.append(
-            global_state.new_bitvec("extcodehash_{}".format(str(address)), 256)
-        )
+        world_state = global_state.world_state
+        stack = global_state.mstate.stack
+        address = Extract(159, 0, stack.pop())
+
+        if address.symbolic:
+            log.debug("unsupported symbolic address for EXTCODEHASH")
+            stack.append(global_state.new_bitvec("extcodehash_" + str(address), 256))
+            return [global_state]
+
+        if address.value not in world_state.accounts:
+            code_hash = symbol_factory.BitVecVal(0, 256)
+        else:
+            code = world_state.accounts_exist_or_load(
+                hex(address.value), self.dynamic_loader
```

Comment:
`hex(address.value)` can sometimes lead to weird, difficult to detect bugs - If the address starts with a 0, like `0x0ab4606496db368e3b0dbbf95d75dad1bb433cbd`, hex(address) will return `0xab4606496db368e3b0dbbf95d75dad1bb433cbd`, sometimes causing parsing errors. Just to be on the safe side, it would be good to pad the hexadecimal part to be exactly 40 characters long.

Reply:
Nice, good idea!